kimchi/src/folding/mod.rs

```diff
@@ -22,13 +22,29 @@ type ScalarField<C> = <<C as FoldingConfig>::Curve as AffineCurve>::ScalarField;
 
 pub trait FoldingConfig: Clone + Debug + Eq + Hash + 'static {
     type Column: FoldingColumnTrait + Debug + Eq + Hash;
+
+    /// The type of an abstract challenge that can be found in the expressions
+    /// provided as constraints.
     type Challenge: Clone + Copy + Debug + Eq + Hash;
+
+    /// The target curve used by the polynomial commitment
     type Curve: CommitmentCurve;
+
     type Srs: SRS<Self::Curve>;
+
+    /// FIXME: use Sponge from kimchi
+    /// The sponge used to create challenges
     type Sponge: Sponge<Self::Curve>;
+
+    /// For Plonk, it will be the commitments to the polynomials and the challenges
     type Instance: Instance<Self::Curve>;
+
+    /// For PlonK, it will be the polynomials in evaluation form that we commit to, i.e. the columns.
+    /// In the generic prover/verifier, it would be WitnessColumns.
```

Comment:
Note: it changes. It is now `Witness`.
```suggestion
    /// In the generic prover/verifier, it would be Witness.
```

It might require some annotations like `kimchi_msm::witness::Witness`.